extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -280,27 +280,16 @@ export class AmpConsent extends AMP.BaseElement {
       this.consentStateManager_.registerConsentInstance(instanceId);
       this.getConsentRemote_(instanceId).then(response => {
         this.parseConsentResponse_(instanceId, response);
-        this.handleUI_(instanceId);
+        this.handlePromptUI_(instanceId);
       }).catch(unusedError => {
         // TODO: Handle errors
       });
     }
-    this.notificationUiManager_.onQueueEmpty(() => {
-      if (!this.revokeUI_) {
-        return;
-      }
-      this.element.classList.add('amp-active');
-      this.element.classList.remove('amp-hidden');
-      setImportantStyles(this.revokeUI_, {display: 'block'});
-    });
 
-    this.notificationUiManager_.onQueueNotEmpty(() => {
-      if (!this.revokeUI_) {
-        return;
-      }
-      this.element.classList.add('amp-hidden');
-      this.element.classList.remove('amp-active');
-      toggle(this.revokeUI_, false);
+    // TODO(@zhouyx): Use setTimeout to make sure we handle postPromptUI
+    // after all prompt UI registerd. Make handle PromptUI a promise instead.
+    this.win.setTimeout(() => {
```

Comment:
Why is the `setTimeout` needed? `this.handlePostPromptUI_()` adds callbacks to the `notificationUiManager_` which don't look like they need to wait a tick for anything. I'm afraid the setTimeout will cause race conditions later. Is there a `Promise` that this code needs to wait for?

Reply:
In theory it should wait for all prompt UI are registered. But that's not a `Promise` yet, I put ad TODO to make it a promise 😢 ... I added `setTimout` to not prompt `postPromptUI` before `promptUI` are handled. What kind of race condition are you thinking? because I don't think this will cause any race condition?

Reply:
😄 haha cool